yasnippet-tests.el

```diff
@@ -614,6 +614,26 @@ foobaz\n"))))
          (should (eq 'global (nth 0 yas--ran-exit-hook)))
          (should (eq 'local (nth 1 yas--ran-exit-hook))))))))
 
+(ert-deftest snippet-mirror-bindings ()
+  "Check that variables defined with the expand-env field is
+accessible with the mirror transformations."
+  (with-temp-buffer
+    (yas-saving-variables
+     (let ((yas-triggers-in-field t)
+           (yas-good-grace nil))
+       (yas-with-snippet-dirs
+         '((".emacs.d/snippets"
+            ("emacs-lisp-mode"
+             ("baz" . "\
+# expand-env: ((func (lambda (str) (upcase str))))
+# --
+hello ${1:$(when (stringp yas-text) (funcall func yas-text))} foo${1:$$(concat \"baz\")}$0"))))
```

Comment:
It is. It's really when we use `${n:$$(...)}` that the problem arises. I could have used `print` instead of `concat`.